src/apps/backend/modules/share-task-request/internal/share-task-request-util.ts

```diff
@@ -0,0 +1,41 @@
+import { ShareTaskRequest } from '../types';
+import { ShareTaskRequestDB } from './store/share-task-request-db';
+import { Task } from '../../task/types';
+import { Account } from '../../account/types';
+import { Types } from 'mongoose';
+
+export default class ShareTaskRequestUtil {
+  public static convertShareTaskDBRequestToShareTaskRequest(
+    shareTaskRequestDb: ShareTaskRequestDB
+  ): ShareTaskRequest {
+    return {
+      id: shareTaskRequestDb._id.toString(),
+      task: this.convertTaskDbToTask(shareTaskRequestDb.task),
+      account: this.convertAccountDbToAccount(shareTaskRequestDb.account),
+    } as ShareTaskRequest;
+  }
+
+  private static convertTaskDbToTask(task: Types.ObjectId | Task): string | Task {
+    return Types.ObjectId.isValid(task.toString())
+      ? task.toString()
+      : {
+          id: (task as Task).id,
+          account: (task as Task).account,
+          description: (task as Task).description,
+          title: (task as Task).title,
+        };
+  }
+
+  private static convertAccountDbToAccount(account: Types.ObjectId | Account): string | Account {
```

Comment:
Is the type of populated `shareTaskRequestDb.account` is `Account`? I don't think so. Please verify.

Reply:
Yes sir!
It's not Account. It is ObjectId only. I corrected this too.